runtime/backtrace_nat.c

```diff
@@ -291,28 +300,56 @@ void caml_debuginfo_location(debuginfo dbg, /*out*/ struct caml_loc_info * li)
   /* Recover debugging info */
   info1 = ((uint32_t *)dbg)[0];
   info2 = ((uint32_t *)dbg)[1];
-  name_info = (struct name_info*)((char *) dbg + (info1 & 0x3FFFFFC));
   /* Format of the two info words:
-       llllllllllllllllllll aaaaaaaa bbbbbbbbbb ffffffffffffffffffffffff k n
-                         44       36         26                        2 1 0
+     Two possible formats based on value of bit 63 (p):
+     Partially packed format (e = 1)
+       |------------ info2 ------------||------------- info1 -------------|
+       elllllllllllllllllll mmmmmmmmmmmmmmmmmm ffffffffffffffffffffffff k n
```

Comment:
```suggestion
      Partially packed format
       |------------ info2 -------------||------------- info1 -------------|    
       1 lllllllllllllllllll mmmmmmmmmmmmmmmmmm ffffffffffffffffffffffff k n
                          44            ||   26                        2 1 0
      Fully packed format:
       |------------- info2 ---------------||------------- info1 -------------|
       0 llllllllllll mmm aaaaaa bbbbbbb ooooooooo ffffffffffffffffffffffff k n
                   51  48     42      35   ||   26                        2 1 0    
```
(and remove the comment explaining field "e" ?

Reply:
Yes, that looks much less unclear!